### PR TITLE
Jiyong's branch

### PR DESCRIPTION
**Pull request by JiyongSong1**
Updated rental history routes in nft.ts

### CODE REVIEW

File: frontend/src/Account.js [hunk 1]
```diff
@@ -15,57 +17,120 @@ function Account() {
                 if (accounts.length > 0) {
                     setAuthenticated(true);
                     setAccount(accounts[0]);
+                    console.log(accounts[0]);
```
Comment (echang49) [inline frontend/src/Account.js:20]: No console.logs

Comment (echang49) [inline frontend/src/Account.js:20]: No console.logs

File: frontend/src/Account.js [hunk 2]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+            axios.get("/api/nft/listings", 
+                {
                     params: {
                         available: true,
                         publicAddress: account,
                     },
                 })
                 .then((res) => {
                     console.log(res.data);
+                    var data = res.data;
```
Comment (echang49) [inline frontend/src/Account.js:40]: use let or const. not var

File: frontend/src/Account.js [hunk 3]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+        if (isAuthenticated) {
+            axios.get("/api/nft/listings", 
+                {
                     params: {
                         available: true,
                         publicAddress: account,
                     },
                 })
                 .then((res) => {
                     console.log(res.data);
```
Comment (echang49) [inline frontend/src/Account.js:39]: console.log

File: frontend/src/Account.js [hunk 4]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
-                    navigate("/login")
+                    alert ("There was an error, please try again");
                 }
             });
     }, []);
 
-    // returns all nfts listed by an account, to be checked later
+    // returns all nfts for the logged in account
     useEffect(() => {
-        if (isAuthenticated && account !== undefined) {
```
Comment (echang49) [inline frontend/src/Account.js:27]: What happens if account is undefined

File: frontend/src/Account.js [hunk 5]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+        }, {
+          params: {
+            publicAddress: publicAddress,
+            signature: signature
+          }
+        })
+        .then((res) => {
+          console.log(res.data)
+        });
+      }
```
Comment (echang49) [inline frontend/src/Account.js:69]: Why is this code here? This is hardcoded.

File: frontend/src/Account.js [hunk 6]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+                                <button id={"button" + index} onClick={() => navigate('/nft?id=' + list._id)}> Rent </button>
+                            </td>
+                        </tr>
+                        )
+                    }
+                </tbody>
+            </table>
+        </div>*/
+
+        /*Eustace's format*/
```
Comment (echang49) [inline frontend/src/Account.js:115]: Why is this code commented?

File: frontend/src/Account.js [hunk 7]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+          }
+        })
+        .then((res) => {
+          console.log(res.data)
+        });
+      }
+
+    async function remove(id) {
+        
+        var answer = window.confirm ("Would you like to delete this listing?");
```
Comment (echang49) [inline frontend/src/Account.js:73]: Is there a better way to implement this?

File: frontend/src/Account.js [hunk 8]
```diff
@@ -15,57 +17,120 @@ function Account() {
...
+                  listingID: id,
+                  signature: signature,
                 }
-            });
-        //const [publicAddress, signature] = await authenticateAction(navigate);
-    }
+              })
+              .then((res) => {
+                console.log(res.data)
+                alert ('Your listing has been deleted.')
```
Comment (echang49) [inline frontend/src/Account.js:85]: Is there a better way to implement this?

File: backend/src/routes/nft.ts [hunk 9]
```diff
@@ -22,6 +22,30 @@ export module NFT {
...
+    });
+
+    /**
+     * PURPOSE: Returns rental history of a single NFT
+     * USAGE: Display past and current NFT rentals onto the user interface
+     */
+    router.get('/nfthistory', async (req: Request, res: Response) => {
+        const { id } = req.query;
+        const rental = await Rentals.find({listingID: id}).lean()
+        if(!rental /*|| rental?.listingID !== id*/)
```
Comment (echang49) [inline backend/src/routes/nft.ts:44]: Remove comment if not needed

**Review by echang49 — CHANGES_REQUESTED** [review]

File: frontend/src/Account.js (hunk 4 above)
Comment (JiyongSong1) [reply frontend/src/Account.js:27]: Added a condition to check that account is defined

File: frontend/src/Account.js (hunk 3 above)
Comment (JiyongSong1) [reply frontend/src/Account.js]: Deleted

File: frontend/src/Account.js (hunk 2 above)
Comment (JiyongSong1) [reply frontend/src/Account.js]: Used let

File: frontend/src/Account.js (hunk 8 above)
Comment (JiyongSong1) [reply frontend/src/Account.js:63]: This is the simplest implementation

File: backend/src/routes/nft.ts (hunk 9 above)
Comment (JiyongSong1) [reply backend/src/routes/nft.ts]: Removed unnecessary comment

**Review by JiyongSong1 — COMMENTED** [review]
Edited the code, request a review

**Review by echang49 — APPROVED** [review]

**Pull request merged**